### PR TITLE
preset passmanagers do not run CXDirection if coupling map is symmentric

### LINKED ISSUE #2183: cx_direction pass should not be run for symmetric coupling_maps
### What is the expected enhancement?
The `cx_direction` pass should not be run if the device `coupling_map` is symmetric.  Currently it is always run.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #2183

I extended `CouplingMap` with `is_symmetric` property. That is use as a condition to run the CXDirection pass. Tests are also added.

### CODE REVIEW

File: qiskit/transpiler/preset_passmanagers/default.py [hunk 1]
```diff
@@ -72,7 +72,9 @@ def default_pass_manager(transpile_config):
     pass_manager.append(Decompose(SwapGate))
 
     # Change CX directions
-    pass_manager.append(CXDirection(coupling_map))
+    pass_manager.append(CXDirection(coupling_map),
+                        condition=lambda property_set: (not coupling_map.is_symmetric and
+                                                        not property_set['is_direction_mapped']))
```
Comment (kdk) [inline qiskit/transpiler/preset_passmanagers/default.py]: Should we want to add a `CheckCXDirection` pass prior?

File: qiskit/transpiler/coupling.py [hunk 2]
```diff
@@ -186,6 +190,27 @@ def shortest_undirected_path(self, physical_qubit1, physical_qubit2):
...
+
+    def _check_symmetry(self):
+        """
+        Calculates symmetry
+
+        Returns:
+            Bool: True if symmetric, False otherwise
+        """
+        mat = nx.adjacency_matrix(self.graph)
+        return (abs(mat - mat.T) > 1e-10).nnz == 0
```
Comment (kdk) [inline qiskit/transpiler/coupling.py]: Is the `1e-10` check required? Edges in `mat` should only be integers (the number of edges between two nodes).

Comment (1ucian0) [reply qiskit/transpiler/coupling.py]: Fixed in 407723d2

File: qiskit/transpiler/preset_passmanagers/default.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/transpiler/preset_passmanagers/default.py]: Yes... also `CheckCXDirection` should only run when `not coupling_map.is_symmetric`. I will do that in a following PR (it is not directly related)...

File: qiskit/transpiler/coupling.py [hunk 3]
```diff
@@ -86,6 +88,7 @@ def add_physical_qubit(self, physical_qubit):
         self.graph.add_node(physical_qubit)
         self._dist_matrix = None  # invalidate
         self._qubit_list = None  # invalidate
+        self._is_symmetric = None  # invalidate
```
Comment (ajavadia) [inline qiskit/transpiler/coupling.py]: not sure why adding a qubit should invalidate symmetry

File: qiskit/transpiler/preset_passmanagers/default.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/transpiler/preset_passmanagers/default.py]: Summitted in https://github.com/Qiskit/qiskit-terra/issues/2362

**Pull request merged**